metriq_gym/schema_validator.py

```diff
@@ -45,17 +45,21 @@ def create_pydantic_model(schema: dict[str, Any]) -> Any:
     return model
 
 
-def load_and_validate(file_path: str, schema_dir: str = DEFAULT_SCHEMA_DIR) -> BaseModel:
-    """
-    Load parameters from a JSON file and validate them against the corresponding schema.
-
-    Raises a ValidationError if validation fails.
-    """
-    params = load_json_file(file_path)
+def validate_model(params: dict[str, Any], schema_dir: str = DEFAULT_SCHEMA_DIR) -> BaseModel:
```

Comment:
It seems odd that `validate_model` would return a model instead of a boolean indicating that the model is valid. Do we just want a method called `load_model` which does the validation within that function? IOW, is there ever a scenario where we load a model that we *don't* want to have it validated as well? Might make sense to combine the validation and load logic into one function.